test/sagemaker_tests/huggingface_tensorflow/training/resources/scripts/train.py

```diff
@@ -11,7 +11,7 @@
 
 if os.environ.get("SDP_ENABLED") or is_sagemaker_dp_enabled():
     SDP_ENABLED = True
-    os.environ["SAGEMAKER_INSTANCE_TYPE"] = "p3dn.24xlarge"
+    os.environ["SAGEMAKER_INSTANCE_TYPE"] = "p4d.24xlarge"
```

Comment:
Why does sagemaker_dp require p4d's?

Reply:
@arjkesh In memory, I once had issues launching sagemaker jobs with p3 instances. It could be just for my AWS account. I set the instance types back. Let's see if the tests pass.

Reply:
I can confirm that the OOM issue is solved with a smaller batch size (tested with 8). Maybe we can use a smaller `batch_size` for the test, and then we can continue to use P3 instances instead of p4d's.